### PR TITLE
[MAINT] Put references in alphabetical order

### LINKED ISSUE #3304: Reorder references alphabetically in BibTeX reference list
Alphabetical order should be restored in reference list `doc/references.bib` to avoid duplication of entries.
Pointed out in https://github.com/nilearn/nilearn/pull/3302#discussion_r915647613

### PR DESCRIPTION
**Pull request by ymzayek**
Closes #3304.

Changes proposed in this pull request:

- Alphabetize `references.bib`

### CODE REVIEW

**Review by tsalo — APPROVED** [review]
LGTM!

File: nilearn/glm/regression.py [hunk 1]
```diff
@@ -342,7 +342,7 @@ def residuals(self):
     def normalized_residuals(self):
         """Residuals, normalized to have unit length.
 
-        See :footcite:`Montgomery` and :footcite:`GVK393847152`.
+        See :footcite:`Montgomery` and :footcite:`Davidson`.
```
Comment (bthirion) [inline nilearn/glm/regression.py:345]: Could we take this opportunity to make the citations more homogeneous, e.g. `Davidson2004` ?

**Review by bthirion — COMMENTED** [review]
Thx for adressing this ! LGTM apart from homogeneity of the keys.

File: nilearn/glm/regression.py (hunk 1 above)
Comment (ymzayek) [reply nilearn/glm/regression.py]: Done!

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Pull request merged**